Update h5LT files with latest flex and Bison available on jelly

lrknox opened the pull request:
@gnuoyd Change to H5LTanalyze.c line 126 eliminated warning "rule could not be matched".  Is the change correct?

lrknox committed:
Modify temporary rpath for testing in java example scripts.

lrknox committed:
Merge branch 'develop' of https://github.com/HDFGroup/hdf5 into develop

lrknox committed:
Merge branch 'develop' of https://github.com/HDFGroup/hdf5 into develop

lrknox committed:
Merge branch 'develop' of https://github.com/HDFGroup/hdf5 into develop

lrknox committed:
Update URL in source file Copyright headers for web copy of COPYING (license) file.

lrknox committed:
Merge branch 'develop' of https://github.com/HDFGroup/hdf5 into develop

lrknox committed:
Add release_docs/code-conventions.md file.

lrknox committed:
Merge branch 'develop' of https://github.com/HDFGroup/hdf5 into develop

lrknox committed:
Add script to test h5py.

lrknox committed:
Fix script error.

lrknox committed:
Add file h5pytest.yml.

lrknox committed:
Merge branch 'develop' of https://github.com/HDFGroup/hdf5 into develop

lrknox committed:
Merge branch 'develop' of https://github.com/HDFGroup/hdf5 into develop

lrknox committed:
Merge branch 'develop' of https://github.com/HDFGroup/hdf5 into h5pytest_initial

lrknox committed:
Test declaration of counter variables in for loops.

github-actions[bot] committed:
Committing clang-format changes

lrknox committed:
Merge pull request #1 from lrknox/h5pytest_initial

H5pytest initial

lrknox committed:
Merge branch 'develop' of https://github.com/HDFGroup/hdf5 into develop

github-actions[bot] committed:
Committing clang-format changes

lrknox committed:
Merge branch 'develop' of https://github.com/HDFGroup/hdf5 into develop

lrknox committed:
Added [] to line 126 of H5LTanalyze.l.
Ran bin/genparser with flex v2.6.4 and Bison v3.0.4 on jelly.

lrknox committed:
Revert "Added [] to line 126 of H5LTanalyze.l."

This reverts commit e4a9cee441efa75b16a8ee030c86189e186dd266.

lrknox committed:
Revert extra commits to match canonical develop.

lrknox committed:
Added [] to H5LTanalyze.l line 126 to eliminate warning "rule could not be
matched".
Ran bin/genparser hl/src with flex v2.6.4 and Bison v3.0.4 on jelly.

lrknox committed:
Added HD to strdup in H5LTparse.y line 338.
Ran bin/genparser hlsrc again.

byrnHDF commented on hl/src/H5LTparse.c:
```diff
@@ -1643,419 +1486,418 @@ yyparse ()
                             /*Adjust size and insert member, consider both member size and offset.*/
                             if(cmpd_stack[csindex].first_memb) { /*reclaim the size 1 temporarily set*/
-                                new_size = H5Tget_size((yyvsp[(1) - (5)].hid)) + (yyvsp[(4) - (5)].ival);
+                                new_size = H5Tget_size((yyvsp[-4].hid)) + (yyvsp[-1].ival);
```
This looks weird! However that is how it is in 1.12.

byrnHDF commented on hl/src/H5LTparse.y:
```diff
@@ -315,7 +315,7 @@ strsize         :       H5T_VARIABLE_TOKEN     {$<ival>$ = H5T_VARIABLE_TOKEN;}
                 |       NUMBER
                 ;
 strpad          :       H5T_STR_NULLTERM_TOKEN {$<ival>$ = H5T_STR_NULLTERM_TOKEN;}
-                |       H5T_STR_NULLPAD_TOKEN  {$<ival>$ = H5T_STR_NULLPAD_TOKEN;}
```
This seems like an issue? Why remove it?

gnuoyd commented:
"rule could not be matched" crops up because line 126 matches newlines and a previous line matches newlines with higher priority. The thing to do is remove line 126 and make sure the tests still pass.

gnuoyd commented on hl/src/H5LTanalyze.l:
```diff
@@ -123,7 +123,7 @@ OPQ_TAG          {return token(OPQ_TAG_TOKEN);}
 ";"              {return token(';');}
 [ \t\n]*         ;
-"\n"             { return 0; }
+"[\n]"             { return 0; }
```
BTW, I am pretty sure this new rule matches a newline in literal square brackets.

Without the double quotes, the rule would match a literal newline, and `flex` should warn that the rule will never match, because line 125 matches newlines, also, with higher priority.

lrknox commented:
Thanks, I’ll try that.

We have a failing test with CMake:  hl_test_lite with this output

Testing         text for string types                                 ERROR: syntax error before "H5T_STR_NULLPAD".

And these errors:
HDF5-DIAG: Error detected in HDF5 (1.13.0) thread 0:
  #000: /mnt/wrk/lrknox/hdf5-1.13.0/src/H5T.c line 2334 in H5Tis_variable_str(): not a datatype
    major: Invalid arguments to routine
    minor: Inappropriate type
HDF5-DIAG: Error detected in HDF5 (1.13.0) thread 0:
  #000: /mnt/wrk/lrknox/hdf5-1.13.0/src/H5Tstrpad.c line 54 in H5Tget_strpad(): not a datatype
    major: Invalid arguments to routine
    minor: Inappropriate type

Trying to figure out that problem.  Fails similarly on both github’s ubuntu and jelly.



From: David Young ***@***.***>
Sent: Tuesday, March 23, 2021 5:02 PM
To: HDFGroup/hdf5 ***@***.***>
Cc: Larry Knox ***@***.***>; Author ***@***.***>
Subject: Re: [HDFGroup/hdf5] Update h5LT files with latest flex and Bison available on jelly (#502)


"rule could not be matched" crops up because line 126 matches newlines and a previous line matches newlines with higher priority. The thing to do is remove line 126 and make sure the tests still pass.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/HDFGroup/hdf5/pull/502#issuecomment-805295685>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADGMWCTPIL56YS7C43BJALDTFEFWDANCNFSM4ZVNZVLA>.

gnuoyd commented:
On Tue, Mar 23, 2021 at 03:44:24PM -0700, Larry Knox wrote:
> Thanks, I’ll try that.
> 
> We have a failing test with CMake:  hl_test_lite with this output
> 
> Testing         text for string types                                 ERROR: syntax error before "H5T_STR_NULLPAD".

Does it have something to do with the deleted line in the .y file that
Allen pointed out?  I wondered if it was just an accident.

Dave

lrknox committed:
Removed line 126 from H5LTanalyze.l - previous line matches.
Added previously removed line 318 of H5LTparse.y to fix hl_test_lite
failure.
Ran bin/genparser.
